src/index.js

```diff
@@ -1,13 +1,42 @@
 "use strict";
 const express = require('express');
 const app = express();
+let products = require('./data/products.json');
+let categories = require('./data/categories.json');
 
 const PORT = 3000;
 
 app.get('/', (request, response) => {
   response.send('Hello World');
 });
 
+//returns object of all products in products.json
+app.get('/products/all', (request, response) => {
+  response.send(JSON.stringify(products));
+});
+
+//returns all products with the specified ID parameter from products.json
+app.get('/products/:id', (request, response) => {
+var choice = request.params.id;
+ for(var i = 0; i < products.products.length; i++) {
+    if(choice == products.products[i].id) {
+      response.send(products.products[i]);
+    }
+  }
+});
+
+//returns all products with the categoryID parameter in the URL as an array
+app.get('/category/:ctyId', (request, response) => {
+  var choice = request.params.ctyId;
+  let a = [];
+  for(var i = 0; i < products.products.length; i++) {
+     if(choice == products.products[i].categoryId) {
```

Comment:
```suggestion
     if(choice === products.products[i].categoryId) {
```
See this [SFO answer](https://stackoverflow.com/questions/359494/which-equals-operator-vs-should-be-used-in-javascript-comparisons#359509).